lib/os/CMakeLists.txt

```diff
@@ -24,6 +24,7 @@ zephyr_sources(
   heap-validate.c
   bitarray.c
   multi_heap.c
+  mem_blocks.c
```

Comment:
If this PR is accepted I would like to see compilation of mem_blocks.c be opt-in and dependent on Kconfig to save build-time for all the builds that don't use it.

I know adding it unconditionally follows prior-art, but we can't fix prior-art without making breaking changes. But for new features we can do this the right way.

Reply:
Added kconfig for the API.